### PR TITLE
[AGW][OAI][MME] Tech Debt 2021-April: reduce compilation warnings for MME compilation

### PR DESCRIPTION
**Pull request by rdefosse**
Signed-off-by: Raphael Defosseux <raphael.defosseux@openairinterface.org>

## Summary

As promised, reducing the number of compilation warnings:

- on RHEL8 build : down from 265 to 23

I splitted the "magma_logging.h" into 2 files.

## Additional Information

- [ ] This change is backwards-breaking

### CODE REVIEW

**Comment by rdefosse** [comment]
Reduction effect:

[PR result](https://jenkins-oai.eurecom.fr/job/MAGMA-MME-production/2078/artifact/build_results_magma_oai_mme.html)

vs

[State of Master 2021/04/29](https://jenkins-oai.eurecom.fr/job/MAGMA-MME-production/2075/artifact/build_results_magma_oai_mme.html)

**Review by ulaskozat — APPROVED** [review]
LGTM

File: lte/gateway/c/oai/common/glogwrapper/glog_logging.cpp [hunk 1]
```diff
@@ -91,7 +91,7 @@ void init_logging(const char* app_name, uint32_t default_verbosity) {
 
   sort(glog_files.begin(), glog_files.end());
   if (glog_files.size() <= 10) return;
-  for (int i = 0; i < glog_files.size() - 10; ++i) {
+  for (long unsigned int i = 0; i < glog_files.size() - 10; ++i) {
```
Comment (ardzoht) [inline lte/gateway/c/oai/common/glogwrapper/glog_logging.cpp]:
```suggestion
  for (unsigned long i = 0; i < glog_files.size() - 10; ++i) {
```

**Review by ardzoht — APPROVED** [review]
minor nit, overall LGTM, thanks for this!

File: lte/gateway/c/oai/common/glogwrapper/glog_logging.cpp (hunk 1 above)
Comment (rdefosse) [reply lte/gateway/c/oai/common/glogwrapper/glog_logging.cpp]: Thx Alex. Did not see your comment before I pushed new commits.

I'll add a new one right away.

**Review by themarwhal — APPROVED** [review]

**Comment by themarwhal** [comment]
> Reduction effect:
> 
> [PR result](https://jenkins-oai.eurecom.fr/job/MAGMA-MME-production/2078/artifact/build_results_magma_oai_mme.html)
> 
> vs
> 
> [State of Master 2021/04/29](https://jenkins-oai.eurecom.fr/job/MAGMA-MME-production/2075/artifact/build_results_magma_oai_mme.html)

Thanks for doing this!!! this warning was bothering me for a while... Was there any difference in compilation time?

**Comment by rdefosse** [comment]
> 
> 
> > Reduction effect:
> > [PR result](https://jenkins-oai.eurecom.fr/job/MAGMA-MME-production/2078/artifact/build_results_magma_oai_mme.html)
> > vs
> > [State of Master 2021/04/29](https://jenkins-oai.eurecom.fr/job/MAGMA-MME-production/2075/artifact/build_results_magma_oai_mme.html)
> 
> Thanks for doing this!!! this warning was bothering me for a while... Was there any difference in compilation time?

@themarwhal Marie,

not really. The build time is not impacted.

But it is easier to spot real warnings when you have a small amount than buried amongst 250+

Once these `tech-debt` PR are merged, I will add the HTML report a detailed view of the compilation warnings.

Raphael

**Review by lionelgo — APPROVED** [review]

**Review by koolzz — APPROVED** [review]

**Review by themarwhal — APPROVED** [review]

**Pull request merged**